mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -141,7 +144,7 @@
           // TODO: We do not support residual evaluation for HIVE and PIG in memory data model yet
           checkResiduals(task);
         }
-        splits.add(new IcebergSplit(conf, task, table.io(), table.encryption()));
+        splits.add(new IcebergSplit(SerializableTable.copyOf(table), conf, task));
```

Comment:
Nit: It looks like this call to `SerializableTable.copyOf(table)` is happening in a loop. Can we pull it out?

Reply:
Good catch, @kbendick! Moved it outside of the loop in #2988.

Reply:
No longer applies, resolving.